sycl/source/detail/queue_impl.cpp

```diff
@@ -293,50 +285,31 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
   // directly. Otherwise, only wait for unenqueued or host task events, starting
   // from the latest submitted task in order to minimize total amount of calls,
   // then handle the rest with piQueueFinish.
-  // TODO the new workflow has worse performance with Level Zero, keep the old
-  // behavior until this is addressed
-  if (!is_host() &&
-      getPlugin().getBackend() == backend::ext_oneapi_level_zero) {
+  bool SupportsPiFinish = !is_host() && MSupportOOO;
```

Comment:
```suggestion
  const bool SupportsPiFinish = !is_host() && MSupportOOO;
```
Minor.

Also, do we really need to known if there is a support for piFinish? Isn't it handled in `addEvent` already?

Reply:
`addEvent` could have also added it for another reason, e.g. unenqueued command.